### ISSUE TITLE
clusterEvents not triggered when markers are populated after initial load (delayed creation)

### ISSUE THREAD

**Issue by janpachol** [angular-ui/angular-google-maps#1129]
It looks like clusterEvents property is not registered when markers array is empty and is populated after  uiGmapIsReady promise is ready.

```javascript
        $scope.map = {
            center: {
                latitude: 50.087743,
                longitude: 14.478982
            },
            zoom: 8,
            options: {
                scrollwheel: true
            },
            clusterEvents: {
                click: function(cluster, clusterModels) {
                    console.log("click cluster");
                },
                mouseout: function() {
                    console.log("mouseout cluster");
                }
            },
            markers: []
        };

        $scope.find = function() {
            $scope.map.markers = [];
            $scope.map.markers.push({
                id: "neco1",
                coordinates: {
                    latitude: 50.087743,
                    longitude: 14.478982
                }
            });
            $scope.map.markers.push({
                id: "neco2",
                coordinates: {
                    latitude: 50.087743,
                    longitude: 14.478982
                }
            });
        };

        $scope.find();
```

This example is OK, but when $scope.find() is called in uiGmapIsReady promise:

```javascript
        uiGmapIsReady.promise(1).then(function(instances) {
            $scope.find();
        });
```

Then clusterEvents are not triggered

I'm adding html for completition:

```HTML
    <ui-gmap-google-map center="map.center" zoom="map.zoom" draggable="true" options="map.options" bounds="map.bounds">
        <ui-gmap-markers models="map.markers" coords="'coordinates'" icon="'icon'" doCluster="true" clusteroptions="map.clusterOptions" clusterevents="map.clusterEvents">
        </ui-gmap-markers>
    </ui-gmap-google-map>

**Comment by nmccready** [comment]
Is isReady really relevant to this? IE: wouldn't `$timeout(someFn, 500)` have the same effect?

**Comment by janpachol** [comment]
Yes, it should have same effect. Sorry I wasn't completely accurate.

**Comment by nmccready** [comment]
So [this](https://github.com/angular-ui/angular-google-maps/blob/master/src/coffee/directives/api/models/parent/markers-parent-model.coffee#L83-L96) is probably the issue

**Comment by nmccready** [comment]
It is definitely some race somewhere. With a $timeout of 250ms it still works, but at a 1000ms it does not.

**Comment by nmccready** [comment]
Resolved by #1135
